pyam/core.py

```diff
@@ -580,6 +589,7 @@ def rename(self, mapping=None, inplace=False, append=False, **kwargs):
         # renaming is only applied where a filter matches for all given columns
         rows = ret._apply_filters(filters)
         idx = ret.meta.index.isin(_make_index(ret.data[rows]))
+        _data = ret.data.copy()
```

Comment:
why do we need to make a copy here?

Reply:
In case `check_duplicates=True` and a `ValueError` is raised, we don't want any changes to the `IamDataFrame` to persist.

But I guess `_data = ret.data.copy() if check_duplicates else ret.data` would be slightly more efficient, right?

Reply:
Agreed. Can you add a warning in the docstring stating that this makes a copy as well? Also, are you sure we want that to be `True` by default? What are the pro/cons?

Reply:
Added the more elaborate only-make-interim-copy-if-`check_duplicates` and an inline comment to explain that the interim copy is kept until after the validation check.

Reply:
> Also, are you sure we want that to be `True` by default? What are the pro/cons?

I think that the more careful behaviour (`check_duplicates == True`) should be the default. Users should explicitly have to override the sanity check when (as in the unit test) renaming timeseries `test_1` to `test_3` *and aggregating* with existing timeseries data of `test_3`.